### ISSUE TITLE
Do not flag Obsolete modules as Unsupported

### ISSUE THREAD

**Issue by janmashat** [mxr576/ddqg#8]
Based on Drupal standards, modules marked as Obsolete may still be supported.

For example based on [this](https://drupal.slack.com/archives/C5B7P7294/p1694520838481369) discussion in Slack, [swiftmailer](https://www.drupal.org/project/swiftmailer) 8.x-2.4 is technically covered by the Drupal Security Team - even if the wording on the project page may indicate otherwise.

**Comment by mxr576** [comment]
:exploding_head: :exploding_head: :exploding_head: 

Another example that challenges the current categorization is https://www.drupal.org/project/variationcache. Since Drupal core 10.2.0 was released, the module was marked as "Obsolete" (also "Minimally maintained"), which should not be categorized as unsupported.

> This feature is [now available in Drupal 10.2](https://www.drupal.org/node/3365546)! 🚀
>
> Please uninstall and remove this module once you are using Drupal 10.2 or higher and have no more code mentioning the Drupal\variationcache namespace.

**Suggested refactoring based on Drupal's FTW maintenance categorization: introduce a new "deprecated" list.**

New rules:
* `Development status === Obsolete` -> List: deprecated
* Version is in an unsupported branch -> List = unsupported (see "supported branches" in https://updates.drupal.org/release-history/config_ignore/current)
* `Maintenance status === Unsupported` -> List = unsupported 


A potential follow-up task: rename "unsupported" to "abandoned".

**Comment by mxr576** [comment]
This is what happened after the split was introduced:

* the unsupported list became more explicit: https://github.com/mxr576/ddqg/commit/e3e7c81676ce675ce432ec8c2795601c1798e6e4
* the deprecated list was born: https://github.com/mxr576/ddqg/commit/5496b213feddb978c28bdc977e130df1f88a4e82

..and God bless missing validation rules on D.o packagist ;S 

```diff
    // @todo Find a better approach for filtering out projects with invalid package names.
    //   conflict.drupal/dummy__common is invalid, it should have a vendor name, a forward slash,
    //   and a package name. The vendor and package name can be words
    //   separated by -, . or _. The complete name should match "^[a-z0-9]([_.-]?[a-z0-9]+)*/[a-z0-9](([_.]?|-{0,2})[a-z0-9]+)*$".
    if ('dummy__common' === $project_name) {
        return $carry;
    }
```
